src/i965_encoder_common.h

```diff
@@ -0,0 +1,517 @@
+ *
+ * Authors:
+ *     Pengfei Qu <Pengfei.Qu@intel.com>
+ *
+ */
+
+#ifndef _I965_COMMON_ENCODER_H
+#define _I965_COMMON_ENCODER_H
+
+#include <drm.h>
+#include <i915_drm.h>
+#include <intel_bufmgr.h>
+
+#include <va/va.h>
+#include "i965_encoder.h"
+#include "i965_gpe_utils.h"
+
+struct encode_state;
+struct intel_encoder_context;
+
+/*
+   this file define the common structure for encoder, such as H264/H265/VP8/VP9
+*/
+#define INTEL_BRC_NONE   0x00000001
+#define INTEL_BRC_CBR    0x00000002
+#define INTEL_BRC_VBR    0x00000004
+#define INTEL_BRC_CQP    0x00000010
+#define INTEL_BRC_AVBR   0x00000040
+
+#define INTEL_BRC_INIT_FLAG_CBR                        0x0010,
+#define INTEL_BRC_INIT_FLAG_VBR                        0x0020,
+#define INTEL_BRC_INIT_FLAG_AVBR                       0x0040,
+#define INTEL_BRC_INIT_FLAG_CQL                        0x0080,
+#define INTEL_BRC_INIT_FLAG_FIELD_PIC                  0x0100,
+#define INTEL_BRC_INIT_FLAG_ICQ                        0x0200,
+#define INTEL_BRC_INIT_FLAG_VCM                        0x0400,
+#define INTEL_BRC_INIT_FLAG_IGNORE_PICTURE_HEADER_SIZE 0x2000,
+#define INTEL_BRC_INIT_FLAG_QVBR                       0x4000,
+#define INTEL_BRC_INIT_FLAG_DISABLE_MBBRC              0x8000
+
+
+#define INTEL_BRC_UPDATE_FLAG_FIELD                  0x01,
+#define INTEL_BRC_UPDATE_FLAG_MBAFF                  (0x01 << 1),
+#define INTEL_BRC_UPDATE_FLAG_BOTTOM_FIELD           (0x01 << 2),
+#define INTEL_BRC_UPDATE_FLAG_ACTUALQP               (0x01 << 6),
+#define INTEL_BRC_UPDATE_FLAG_REFERENCE              (0x01 << 7)
+
+#define INTEL_VME_MIN_ALLOWED_WIDTH_HEIGHT           48
+
+extern const unsigned int table_enc_search_path[2][8][16];
+
+// BRC Flag in BRC Init Kernel
+typedef enum _INTEL_ENCODE_BRCINIT_FLAG
+{
+    INTEL_ENCODE_BRCINIT_ISCBR                       = 0x0010,
+    INTEL_ENCODE_BRCINIT_ISVBR                       = 0x0020,
+    INTEL_ENCODE_BRCINIT_ISAVBR                      = 0x0040,
+    INTEL_ENCODE_BRCINIT_ISCQL                       = 0x0080,
+    INTEL_ENCODE_BRCINIT_FIELD_PIC                   = 0x0100,
+    INTEL_ENCODE_BRCINIT_ISICQ                       = 0x0200,
+    INTEL_ENCODE_BRCINIT_ISVCM                       = 0x0400,
+    INTEL_ENCODE_BRCINIT_IGNORE_PICTURE_HEADER_SIZE  = 0x2000,
+    INTEL_ENCODE_BRCINIT_ISQVBR                      = 0x4000,
+    INTEL_ENCODE_BRCINIT_DISABLE_MBBRC               = 0x8000
+} INTEL_ENCODE_BRCINIT_FLAG;
+
+// BRC Flag in BRC Update Kernel
+typedef enum _INTEL_ENCODE_BRCUPDATE_FLAG
+{
+    INTEL_ENCODE_BRCUPDATE_IS_FIELD                  = 0x01,
+    INTEL_ENCODE_BRCUPDATE_IS_MBAFF                  = (0x01 << 1),
+    INTEL_ENCODE_BRCUPDATE_IS_BOTTOM_FIELD           = (0x01 << 2),
+    INTEL_ENCODE_BRCUPDATE_IS_ACTUALQP               = (0x01 << 6),
+    INTEL_ENCODE_BRCUPDATE_IS_REFERENCE              = (0x01 << 7)
+} INTEL_ENCODE_BRCUPDATE_FLAG;
+
+/*
+kernel operation related defines
+*/
```

Comment:
The following two definitions still include a lot of definitions that is not used.
For example: MBENC_WIDI, VDENC_ME, RESET_VLINE_STRIDE and so on.,
Please double check it and remove the unused/unmeaningful definition.